microk8s-resources/actions/common/utils.sh

```diff
@@ -219,6 +219,46 @@ skip_opt_in_config() {
     fi
 }
 
+# Function to sanitize arguments for API server
+sanatize_args_kubeapi_server() {
+  local deprecated_args=("insecure-port" "feature-gates=RemoveSelfLink")
+
+  # Set the service name
+  local service_name="kube-apiserver"
+
+  for arg in "${deprecated_args[@]}"; do
+    echo "Removing argument: $arg from $service_name"
+    skip_opt_in_local_config "$arg" "$service_name"
+  done
+}
+
+# Function to sanitize arguments for kubelet
+sanatize_args_kubelet() {
+  local deprecated_args=("log-dir")
+
+  # Set the service name
+  local service_name="kubelet"
+
+  for arg in "${deprecated_args[@]}"; do
+    echo "Removing argument: $arg from $service_name"
+    skip_opt_in_local_config "$arg" "$service_name"
+  done
+}
+
+# Function to sanitize arguments for kube-proxy
+sanatize_args_kube_proxy() {
+  local deprecated_args=("proxy-mode")
```

Comment:
Since we remove this argument here we should not handle it speperately in the kubelite wrapper.

Reply:
That is checking for a particular mode for proxy (--proxy-mode=userspace). Is it fine to remove that?